### ISSUE TITLE
Rocky9.2 - could not delete networks: plugin type="portmap" failed (delete): neither iptables nor ip6tables usable

### ISSUE THREAD

**Issue by narein257** [apptainer/apptainer#1362]
### Version of Apptainer
apptainer version 1.1.8

### Expected behavior

Expected to start nginx web server by running
apptainer run --fakeroot --net --network-args="portmap=8080:80/tcp" --writable-tmpfs nginx.sif

### Actual behavior

WARNING: Only --network=fakeroot is permitted in --fakeroot mode. You requested 'bridge'.
WARNING: Overriding with --network=fakeroot
INFO:    unknown argument ignored: lazytime
unknown argument ignored: xino=on
ERROR:   could not delete networks: plugin type="portmap" failed (delete): neither iptables nor ip6tables usable
FATAL:   container creation failed: plugin type="ptp" failed (add): failed to locate iptables: could not get iptables version: exit status 111

### Steps to reproduce this behavior

1. Build apptainer from source with suid - ./mconfig --with-subuid
2. Set fakeroot privilages to user id
3. Build container image - apptainer build nginx.sif docker://nginx
4. Run nginx.sif

### What OS/distro are you running

NAME="Rocky Linux"
VERSION="9.2 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.2"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.2 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.2"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.2"

### How did you install Apptainer
Built from source --with-suid option.

**Comment by narein257** [comment]
![image](https://github.com/apptainer/apptainer/assets/87361149/6f3dbdbb-1e9e-45e3-8f46-81702b2d76ca)

**Comment by DrDaveD** [comment]
Does it work with real root?  I'm not sure if fakeroot has the privileges.

**Comment by narein257** [comment]
Yes, I can start the nginx with sudo apptainer run --net --network-args="portmap=8080:80/tcp" --writable-tmpfs nginx.sif. The problem is don't have sudo privilages in HPC. I need to be able to run nginx without sudo on HPC. So I am testing with fakeroot on my vm before I move this to HPC. 
I am able to run this sucessfully with aptainer v1.0.3 on Rocky8.7. But on Rocky 9 both v1.0.3 and v1.1.8 fail with the iptables error

**Comment by narein257** [comment]
I tried with apptainer 1.1.8 on Rocky 8.7. I was successful in starting nginx with writable sandbox. But when I used sif image with a overlay I got an error -
2023/06/01 06:40:43 [emerg] 44707#44707: mkdir() "/var/cache/nginx/client_temp" failed (61: No data available)
nginx: [emerg] mkdir() "/var/cache/nginx/client_temp" failed (61: No data available)

![image](https://github.com/apptainer/apptainer/assets/87361149/1bfa9f39-b8c4-48c2-8207-8d942a608258)

**Comment by DrDaveD** [comment]
Wait, without apptainer-suid installed on Rocky 9.1 I get those messages:
```
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
...
2023/06/06 20:09:47 [emerg] 5464#5464: mkdir() "/var/cache/nginx/client_temp" failed (61: No data available)
nginx: [emerg] mkdir() "/var/cache/nginx/client_temp" failed (61: No data available)
```
but then I also get
```
WARNING: fakeroot with unprivileged installation or 'allow setuid = no' could not use 'fakeroot' network, fallback to 'none' network
```

However I do not see those messages without apptainer-suid on Rocky 8.8.

Do you have apptainer-suid installed on your systems?  Are there any non-default settings on your 8.7 and 9.2 systems?

**Comment by DrDaveD** [comment]
Oh yes I missed that your original description said you compiled with `--with-suid`.  That should be equivalent to installing the apptainer-suid rpm and in fact you do see the same behavior. 

I am still having no luck reproducing the Rocky8 issue.  Anyway we're getting off the track of this issue which is an el9 issue.  I suggest that you report your experience with "No Data Available" instead in #1387, including the output from `apptainer -d`.

**Comment by narein257** [comment]
@DrDaveD any updates on this item? any next steps?

**Comment by DrDaveD** [comment]
I think it must be a difference in the interface to iptables/ip6tables on el9.  I was hoping that @cclerget could take a look.

**Comment by DrDaveD** [comment]
Meanwhile, I think if you can avoid using the `--net` option I would recommend that.  It only works in suid mode, and we usually recommend disabling network namespaces when possible because most CVEs related to unprivileged user namespaces are only exploitable in combination with network namespaces.  If it's possible to modify the container to change the application to listen on port 8080 in the first place, you'll probably be better off.

**Comment by narein257** [comment]
Thanks for the suggestion. Yes, we are going in that direction.

**Comment by DrDaveD** [comment]
I believe this error is coming from `/usr/libexec/apptainer/cni/ptp` (I see it in the output of `strings` on that) but I'm not sure where the source code is for that.  I would need to do more digging, or hopefully @cclerget knows what to do.

**Comment by cclerget** [comment]
@narein257 Fix #1520 on its way for 1.2.0 release

**Comment by narein257** [comment]
Great. Thanks.
